Scripts/Runtime/Entities/Tasks/AbstractTaskSystem.cs

```diff
@@ -1,246 +1,236 @@
     /// </summary>
-    public abstract partial class AbstractTaskSystem : AbstractAnvilSystemBase,
-                                                       ITaskSystem
+    public abstract partial class AbstractTaskSystem<TTaskDriver, TTaskSystem> : AbstractAnvilSystemBase,
+                                                                                 ITaskSystem
+        where TTaskDriver : AbstractTaskDriver<TTaskDriver, TTaskSystem>
+        where TTaskSystem : AbstractTaskSystem<TTaskDriver, TTaskSystem>
     {
-        private static readonly Type I_PROXY_DATA_STREAM_TYPE = typeof(IProxyDataStream);
-
-        //TODO: Enable TaskDrivers again, a Task System can only have one type of TaskDriver since it governs them
-        // private readonly List<TTaskDriver> m_TaskDrivers;
+        private readonly List<TTaskDriver> m_TaskDrivers;
         private readonly ByteIDProvider m_TaskDriverIDProvider;
-        private readonly HashSet<IProxyDataStream> m_PendingDataStreams;
-        private readonly List<ISystemTaskProcessor> m_SystemTaskProcessors;
-        private readonly byte m_SystemLevelContext;
+        private readonly TaskFlowGraph m_TaskFlowGraph;
+
+        private Dictionary<TaskFlowRoute, BulkJobScheduler<JobConfig>> m_SystemJobConfigBulkJobSchedulerLookup;
+        private Dictionary<TaskFlowRoute, BulkJobScheduler<JobConfig>> m_DriverJobConfigBulkJobSchedulerLookup;
 
-        private NativeArray<JobHandle> m_ProcessorDependenciesScratchPad;
+        private BulkJobScheduler<AbstractProxyDataStream> m_SystemDataStreamBulkJobScheduler;
+        private BulkJobScheduler<AbstractProxyDataStream> m_DriverDataStreamBulkJobScheduler;
 
-#if ENABLE_UNITY_COLLECTIONS_CHECKS
-        private bool m_HasCheckedDataStreamUpdateJobsIntegrity;
-#endif
+        private bool m_IsHardened;
+
+        public byte Context
+        {
+            get;
+        }
 
         protected AbstractTaskSystem()
         {
-            // m_TaskDrivers = new List<TTaskDriver>();
+            m_TaskDrivers = new List<TTaskDriver>();
 
             m_TaskDriverIDProvider = new ByteIDProvider();
-            m_SystemLevelContext = m_TaskDriverIDProvider.GetNextID();
-
-            m_PendingDataStreams = new HashSet<IProxyDataStream>();
-            m_SystemTaskProcessors = new List<ISystemTaskProcessor>();
+            Context = m_TaskDriverIDProvider.GetNextID();
 
-            CreateProxyDataStreams();
+            //TODO: Talk to Mike about this. The World property is null for the default world because systems are created via Activator.CreateInstance.
+            //TODO: They don't go through the GetOrCreateSystem path. Is this the case for other worlds? Can we assume a null World is the default one?
+            World currentWorld = World ?? World.DefaultGameObjectInjectionWorld;
+            m_TaskFlowGraph = currentWorld.GetOrCreateSystem<TaskFlowDataSystem>().TaskFlowGraph;
```

Comment:
I know you ran into how this all works with the multi world stuff. Any thoughts?

Reply:
World isn't set until `OnCreate()`
`World.AddSystem_OnCreate_Internal` is where the World reference gets passed into a system

Reply:
Ok thanks, I've added a TODO to fix this.